aws/resource_aws_sfn_state_machine.go

```diff
@@ -125,6 +124,32 @@ func resourceAwsSfnStateMachineRead(d *schema.ResourceData, meta interface{}) er
 	return nil
 }
 
+func resourceAwsSfnStateMachineUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sfnconn
+	log.Printf("[DEBUG] Updating Step Function State Machine: %s", d.Id())
+
+	params := &sfn.UpdateStateMachineInput{
+		StateMachineArn: aws.String(d.Id()),
+		Definition:      aws.String(d.Get("definition").(string)),
+		RoleArn:         aws.String(d.Get("role_arn").(string)),
+	}
+
+	_, err := conn.UpdateStateMachine(params)
+
+	if err != nil {
+
+		if awserr, ok := err.(awserr.Error); ok {
+			if awserr.Code() == "NotFoundException" || awserr.Code() == "StateMachineDoesNotExist" {
+				d.SetId("")
+				return nil
```

Comment:
In place of these 2 lines, it would be preferable to return an error, as in:

```
return fmt.Errorf("Error updating Step Function State Machine: %s", err)
```

Setting an ID to empty removes it from the state, step done by the Read part, always happening before updating/deleting